site/oauth2.md

```diff
@@ -0,0 +1,464 @@
+- `vhost`: This is the virtual host we are granting access to. It also a wildcard pattern. If not specified, `*` will be used.
+- `queue`|`exchange`: queue or exchange name pattern. The location grants the permission to a set of queues (or exchanges) that match it. One location can only specify either `queue` or `exchange` but not both. If not specified, `*` will be used
+- `routing-key`: this is the routing key pattern the location grants the permission to. If not specified, `*` will be used
+
+For more information about wildcard patterns, check the section [Scope-to-Permission Translation](#scope-to-permission-translation).
+
+#### Actions field  
+
+The `actions` field can be either a string containing a single action or a Json array containing zero or many actions.
+
+The supported actions map to either [RabbitMQ permissions](https://www.rabbitmq.com/access-control.html#authorisation):
+
+- `configure`
+- `read`
+- `write`
+
+Or RabbitMQ user tags:
+
+- `administrator`
+- `monitoring`
+- `management`
+- `policymaker`
+
+#### Rich-Permission to Scope translation
+
+Rich Authorization Request permissions are translated into JWT token scopes that use the
+aforementioned convention using the following algorithm:
+
+For each location found in the `locations` where the `cluster` attribute matches the current RabbitMQ server's `resource_server_id`:
+
+  - For each location found in the `locations` field where the `cluster` attribute matches the current RabbitMQ node's `resource_server_id`, the plugin extracts the `vhost`, `queue` or `exchange` and `routing_key` attributes from the location. If the location does not  have any of those attributes, the default value of `*` is assumed. Out of those values, the following scope suffix will be produced:
+    <pre class="lang-ini">scope_suffix = &lt;vhost>/&lt;queue>|&lt;exchange>/&lt;routing-key></pre>
+
+  - For each action found in the `actions` field:
+
+    if the action is not a known user tag, the following scope is produced out of it:
+    <pre class="lang-ini">
+      scope = &lt;resource_server_id>.&lt;action>:&lt;scope_suffix>
+    </pre>
+
+    For known user tag actions, the following scope is produced:
+    <pre class="lang-ini">
+      scope = &lt;resource_server_id>.&lt;action>
+    </pre>
+
+
+The plugin produces permutations of all `actions` by  all `locations` that match the node's configured `resource_server_id`.
+
+In the following RAR example
+<pre class="lang-javascript">
+{
+  "authorization_details": [
+    { "type" : "rabbitmq",  
+      "locations": ["cluster:finance/vhost:primary-*"],
+      "actions": [ "read", "write", "configure"  ]
+    },
+    { "type" : "rabbitmq",
+      "locations": ["cluster:finance", "cluster:inventory" ],
+      "actions": ["administrator" ]
+    }
+  ]
+}
+</pre>
+
+if RabbitMQ node's `resource_server_id` is equal to `finance`, the plugin will compute the following sets of scopes:
```

Comment:
if a RabbitMQ node's `resource_server_id` is equal to `finance`, the plugin will compute the following sets of scopes: